src/web/app/pages-instructor/instructor-course-student-edit-page/instructor-course-student-edit-page.component.ts

```diff
@@ -188,15 +189,19 @@ export class InstructorCourseStudentEditPageComponent implements OnInit, OnDestr
       isSessionSummarySendEmail: shouldResendPastSessionLinks,
     };
 
+    this.isFormSaving = true;
+
     this.studentService.updateStudent({
       courseId: this.courseId,
       studentEmail: this.student.email,
       requestBody: reqBody,
     })
```

Comment:
nit: recommend using `.pipe(finalize(() => this.isFormSaving = false))` instead of two separate duplicate lines below.

Reply:
Got it, made the change. Is there a reason why pipe instead of using the complete arg in `subscribe`?

```javascript
.subscribe((resp: MessageOutput) => {
        this.navigationService.navigateWithSuccessMessage(this.router, '/web/instructor/courses/details',
            resp.message, { courseid: this.courseId });
      }, (resp: ErrorMessageOutput) => {
        this.statusMessageService.showErrorToast(resp.error.message);
      }, () => {
		// Complete argument
        this.isFormSaving = false;
      });
```

Reply:
Complete block does not execute under error conditions.